items/metadata/itemchannellink.js

```diff
@@ -0,0 +1,139 @@
+ */
+const getItemChannelLink = function (itemName, channelUID) {
+  log.debug(`Getting ItemChannelLink ${itemName} -> ${channelUID} from provider...`);
+  const itemChannelLink = managedItemChannelLinkProvider.get(itemName + ' -> ' + channelUID);
+  if (itemChannelLink === null || itemChannelLink === undefined) return null;
+  return {
+    itemName: itemChannelLink.getItemName().toString(),
+    channelUID: itemChannelLink.getLinkedUID().toString(),
+    configuration: utils.javaMapToJsObj(itemChannelLink.getConfiguration().getProperties())
+  };
+};
+
+/**
+ * Adds a new ItemChannelLink to the provider. Therefore, it adds a channel link to an Item.
+ *
+ * @private
+ * @param {string} itemName the name of the Item
+ * @param {string} channelUID
+ * @param {object} [conf] channel configuration
+ * @returns {{itemName: string, configuration: *, channelUID: string}} the ItemChannelLink
+ */
+const _addItemChannelLink = function (itemName, channelUID, conf) {
+  log.debug(`Adding ItemChannelLink ${itemName} -> ${channelUID} to provider...`);
+  const itemChannelLink = _createItemChannelLink(itemName, channelUID, conf);
+  managedItemChannelLinkProvider.add(itemChannelLink);
+  return {
+    itemName: itemChannelLink.getItemName().toString(),
+    channelUID: itemChannelLink.getLinkedUID().toString(),
+    configuration: utils.javaMapToJsObj(itemChannelLink.getConfiguration().getProperties())
+  };
+};
+
+/**
+ * Update an ItemChannelLink in the provider. Therefore, it updates the channel link of an Item.
+ *
+ * @private
+ * @param {string} itemName the name of the Item
+ * @param {string} channelUID
+ * @param {object} [conf] channel configuration
+ * @returns {{itemName: string, configuration: *, channelUID: string}|null} the old ItemChannelLink or `null` if none exists
+ */
+const _updateItemChannelLink = function (itemName, channelUID, conf) {
+  log.debug(`Updating ItemChannelLink ${itemName} -> ${channelUID} in provider...`);
+  const itemChannelLink = _createItemChannelLink(itemName, channelUID, conf);
+  managedItemChannelLinkProvider.update(itemChannelLink);
+  if (itemChannelLink === null || itemChannelLink === undefined) return null;
+  return {
+    itemName: itemChannelLink.getItemName().toString(),
+    channelUID: itemChannelLink.getLinkedUID().toString(),
+    configuration: utils.javaMapToJsObj(itemChannelLink.getConfiguration().getProperties())
+  };
+};
+
+/**
+ * Adds or updates an ItemChannelLink.
+ *
+ * @memberof items.metadata.itemchannellink
+ * @param {string} itemName the name of the Item
+ * @param {string} channelUID
+ * @param {object} [conf] channel configuration
+ * @returns {boolean} `true` if the channel link was added, `false` if it was updated
+ */
+const replaceItemChannelLink = function (itemName, channelUID, conf) {
+  const existing = getItemChannelLink(itemName, channelUID);
+  if (existing === null) {
+    _addItemChannelLink(itemName, channelUID, conf);
+    return true;
```

Comment:
I think returning the previous channel link or null would be better than a boolean ( i try to avoid such patterns).  Again, Java Maps ad other objects tend to do the same thing (previous value or null);